app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/meta/AggregateMetadataHandler.java

```diff
@@ -47,34 +47,38 @@ public boolean canHandle(StepKind kind) {
 
     @Override
     public DynamicActionMetadata handle(DynamicActionMetadata metadata) {
-        try {
-            DataShape outputShape = metadata.outputShape();
-            DataShape inputShape = metadata.inputShape();
+        DataShape outputShape = metadata.outputShape();
+        DataShape inputShape = metadata.inputShape();
 
+        try {
             if (metadata.outputShape() != null) {
                 outputShape = adaptOutputShape(new DataShape.Builder()
-                                                        .createFrom(metadata.outputShape())
-                                                        .decompress()
-                                                        .build());
+                        .createFrom(metadata.outputShape())
+                        .decompress()
+                        .build());
             }
+        } catch (IOException e) {
```

Comment:
As far as I understand the only case in which IOException is thrown is if the data shape's `specification` can't be parsed as JSON or JSON Schema. I'm fine with completely blowing up rather than silently (apart from those WARN logs), as it is a violation of the contract we have. Not sure if this not resulting in an error will end up being an issue for the UI and the integration.

Reply:
The initial reason for me to not blow up on that error was that the call is just a meta data lookup for data shapes. If that call is failing for the mentioned reason that we can not inspect the specification the original specification is returned and the UI should be able to continue. In most cases the integration is able to operate even if this meta data lookup happens to fail.

Maybe come back to that and think over when more lookups are about to come.

Reply:
The risk I'm thinking about is when we provide the wrong (default) data shape to the mapper and this results in an exception at runtime from the mapper.
But really, it should be a rare occurrence and to fix this properly we should have a test harness for connectors that checks the shapes and validations for user provided shapes, so I'm good either way for now.